lib/statsd.rb

```diff
@@ -14,14 +14,77 @@
 #   statsd = Statsd.new('localhost').tap{|sd| sd.namespace = 'account'}
 #   statsd.increment 'activate'
 class Statsd
-  class RubyUdpClient
-    attr_reader :key, :sock
+  class BufferedUDPClient < UDPClient
+    @max_packet_size = 511
+    (class << self; attr_accessor :max_packet_size; end)
 
-    def initialize(address, port, key = nil)
+    def initialize(address, port)
+      super(address, port)
+      @buffer = String.new
+    end
+
+    def flush!
+      return unless @buffer.bytesize > 0
+      sock.write(@buffer)
+      @buffer.clear
```

Comment:
It [looks like `String#clear` deallocates the buffer](https://github.com/ruby/ruby/blob/9efd4dbe73babbc1be41d4bf72db2dbbcb1cfa5d/string.c#L1841-L1850). Is there a way we can keep the buffer around?